match/lib/match/runner.rb

```diff
@@ -120,7 +120,14 @@ def fetch_provisioning_profile(params: nil, certificate_id: nil, app_identifier:
       profile = profiles.last
 
       if params[:force_for_new_devices] && !params[:readonly]
-        params[:force] = device_count_different?(profile: profile) unless params[:force]
+        if prov_type != :appstore
+          params[:force] = device_count_different?(profile: profile) unless params[:force]
+        else
+          # App Store provisioning profiles don't contain device identifiers and
+          # thus shouldn't be renewed if the device count has changed.
+          UI.important "Warning: `force_for_new_devices` is set but is ignored for App Store provisioning profiles."
```

Comment:
Is this really necessary? Why not just silently ignore it?
Other then that LGTM.

Reply:
I agree that it might not be totally necessary, but I do like that it lets people know not to add no-op params to tools because if something changes down the road we will always have to be aware of that. 

That being said, I think this could be moved into a `verify_block` on the option itself and it can print the same message. 🚀

Reply:
@ohayon: Cool, I didn't know about `verify_block`! However, I'm not entirely sure of how I can access the specified provisioning type from the block. Any pointers? 👉

Reply:
Also, if it's moved into the `verify_block`, should it still be a `UI.important` or should it be a `UI.user_error!`? I believe making it an error is required if we want to avoid the nested conditionals (like you mentioned below), but I guess it would also make this a breaking change compared to previous versions of Match? 💣

Reply:
Please don't make it error! 💣 That will break our lanes for updating provision profiles.
Since then you forcing us to also care about the conditional parameter, which we shouldn't have to.

```
  lane :update do |options|
    match(
      type: options[:type],
      git_url: 'ssh://git@bitbucket.widex.com/swd/certificates.git',
      git_branch: 'widex',
      app_identifier: [
        'com.widex.app',
        'com.widex.app.watchkitapp',
        'com.widex.app.watchkitapp.watchkitextension',
        'com.widex.otherapp',
        'com.widex.otherapp.watchkitapp',
        'com.widex.otherapp.watchkitapp.watchkitextension',
        'com.widex.tinnitus'
      ]
    )
  end

  lane :updateAll do
    update(type: 'development')
    update(type: 'adhoc')
    update(type: 'appstore')
  end
```

Reply:
@casz: Yeah, I'm with you. Making it an error feels a bit strict 🙂 Hopefully @ohayon agrees 😃